polyfill/test/monthday.mjs

```diff
@@ -131,6 +131,10 @@ describe('MonthDay', () => {
         throws(() => MonthDay.from('01-15').with({ day: 1 }, { disambiguation }), RangeError)
       );
     });
+    it('cannot lead to an out-of-range MonthDay', () => {
+      const md = MonthDay.from('01-01');
+      equal(`${md.with({ month: 999999 * 12 }, { disambiguation: 'balance' })}`, '12-01');
```

Comment:
In this case would it make a difference if `"constrain"` was used instead?

Reply:
I think it would make a difference; using constrain to create a Date can never be out of range anyway. So I think this test can only use balance.